tests/rptest/tests/datalake/utils.py

```diff
@@ -5,8 +5,22 @@
 #
 # As of the Change Date specified in that file, in accordance with
 # the Business Source License, use of this s
-from rptest.services.redpanda import CloudStorageType, get_cloud_storage_type
+from rptest.services.redpanda import CloudStorageType, get_cloud_provider, get_cloud_storage_type
+from itertools import product
+
+# gcp also uses S3 as storage type
+# Currently we only run tests in ec2 + S3 combination
+# as the query engine deployment was tested only in ec2.
+# The credential plumbing (IAM roles, credential providers) probably
+# needs some fixing in GCP environment before enabling it here, otherwise
+# there no fundamental reason it shouldn't work.
+ACCEPTED_CLOUD_PROVIDERS = {"docker", "aws"}
+ACCEPTED_STORAGE_TYPES = [CloudStorageType.S3]
 
 
 def supported_storage_types():
-    return get_cloud_storage_type(applies_only_on=[CloudStorageType.S3])
+    cloud_provider = ACCEPTED_CLOUD_PROVIDERS.intersection(
+        {get_cloud_provider()})
+    storage_type = get_cloud_storage_type(
+        applies_only_on=ACCEPTED_STORAGE_TYPES)
+    return list(product(cloud_provider, storage_type))
```

Comment:
I stared at this for a while and found the product a bit confusing... Is the intent here different from doing something like below?

```
if get_cloud_provider() not in ACCEPTED_CLOUD_PROVIDERS:
    return []
return get_cloud_storage_type(applies_only_on=ACCEPTED_STORAGE_TYPES)
```

Reply:
Its the same but the approach in the PR shows environment(aws/gcp/docker) in the params output..easier to group the failures visually.

> 'injected_args': {'storage_type': ('docker', <CloudStorageType.S3: 1>